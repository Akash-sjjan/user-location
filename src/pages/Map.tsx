import React, { useContext, useState } from "react";
import { GoogleMap, InfoWindow, Marker, useJsApiLoader } from "@react-google-maps/api";
import Context from "../Context/ContextProvider";

const key = "";  /* Enter Google Map API Key*/

const Map = () => {
  const context: any = useContext(Context);

  const handleOnLoad = (map: any) => {
    const bounds = new google.maps.LatLngBounds();
    context.locationDetails.forEach(({ geometry }: any) => bounds.extend(geometry));
    map.fitBounds(bounds);
  };

  const { isLoaded } = useJsApiLoader({
    googleMapsApiKey: key,
  });
  if (!isLoaded) {
    return <div>Loading.....</div>;
  }
  return (
    <div className="App">
      <GoogleMap onLoad={handleOnLoad} zoom={15} mapContainerStyle={{ width: "100%", height: "100vh" }}>
        {context.locationDetails.map((data: any, index: any) => (
          <Marker key={index} position={data.geometry} label={(index + 1).toString()}></Marker>
        ))}
      </GoogleMap>
    </div>
  );
};

export default Map;
