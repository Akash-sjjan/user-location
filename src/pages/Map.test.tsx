import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import Map from "./Map";
import { GoogleMap, Marker, MarkerProps } from "@react-google-maps/api";
import { mount, shallow } from "enzyme";

let instance: google.maps.Circle | null;

function getMarker(props: MarkerProps) {
  return (
    <GoogleMap>
      <Marker {...props} />
    </GoogleMap>
  );
}
const center = {
  lat: 0,
  lng: 0,
};

const Comp = () => {
  return (
    <BrowserRouter>
      <Map />
    </BrowserRouter>
  );
};
describe("Circle", () => {
  let wrapper = shallow(
    <BrowserRouter>
      <Map />
    </BrowserRouter>
  );
  it("Should render Map Comp correctly", () => {
    render(<Comp />);
  });
  it("should call onLoad only once", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
