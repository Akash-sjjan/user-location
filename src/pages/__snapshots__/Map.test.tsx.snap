// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle should call onLoad only once 1`] = `ShallowWrapper {}`;
